dev/validate_bom__config.py

```diff
@@ -899,6 +913,179 @@ def add_files_to_upload(self, options, file_set):
+
+  def init_argument_parser(self, parser, defaults):
+    """Implements interface."""
+    add_parser_argument(
+        parser, 'pubsub_google_project', defaults, None,
+        help='The name of the GCE project the subscription lives in.')
+    add_parser_argument(
+        parser, 'pubsub_google_credentials_path', defaults, None,
+        help='Path to service account credentials with access to the subscription.')
+    add_parser_argument(
+        parser, 'pubsub_google_subscription_name', defaults, None,
+        help='The name of the pub/sub subscription to pull from.')
+    add_parser_argument(
+        parser, 'pubsub_google_name', defaults, None,
+        help='The logical name of the configured subscription in Spinnaker.')
+    add_parser_argument(
+        parser, 'pubsub_google_template_path', defaults, None,
+        help='Path to the Jinja message translation template.')
+
+  def validate_options(self, options):
+    """Implements interface."""
+    if (options.pubsub_google_project is None
+        or options.pubsub_google_credentials_path is None
+        or options.pubsub_google_subscription_name is None
+        or options.pubsub_google_template_path is None):
+      raise ValueError('Incomplete google pubsub specification. Received:'
+                       ' --pubsub_google_project="{}"'
+                       ' --pubsub_google_credentials_path="{}"'
+                       ' --pubsub_google_subscription_name="{}"'
+                       ' --pubsub_google_template_path="{}"'
+                       .format(options.pubsub_google_project,
+                               options.pubsub_google_credentials_path,
+                               options.pubsub_google_subscription_name,
+                               options.pubsub_google_template_path))
+    if (not os.path.exists(options.pubsub_google_credentials_path)
+        or not os.path.exists(options.pubsub_google_template_path)):
+      raise ValueError('Invalid path specified for one of:'
+                       ' --pubsub_google_credentials_path="{}"'
+                       ' --pubsub_google_template_path="{}"'
+                       .format(options.pubsub_google_credentials_path, options.pubsub_google_template_path))
+    options.pubsub_google_enabled = options.pubsub_google_subscription_name is not None
+
+  def add_config(self, options, script):
+    """Implements interface."""
+    name = options.pubsub_google_name or None
+    project = options.pubsub_google_project or None
+    credentials_path = os.path.basename(options.pubsub_google_credentials_path) or None
+    subscription = options.pubsub_google_subscription_name or None
+    template_path = os.path.basename(options.pubsub_google_template_path) or None
+
+    if name is None:
+      return
+
+    script.append('hal -q --log=info config pubsub google enable')
+    script.append('hal -q --log=info config pubsub google subscription'
+                  ' add {name}'
+                  ' --project {project}'
```

Comment:
I dont think the None values will work here -- will they?
I think instead you want to add the parameters to the call if they are specified.

Reply:
Done.